src/WinRT.Runtime/ActivationFactory.cs

```diff
@@ -0,0 +1,272 @@
+#if NET
+            if (moduleHandle == IntPtr.Zero)
+            {
+                NativeLibrary.TryLoad(fileName, Assembly.GetExecutingAssembly(), null, out moduleHandle);
+            }
+#endif
+            if (moduleHandle == IntPtr.Zero)
+            {
+                module = null;
+                return false;
+            }
+
+            void* getActivationFactory = null;
+
+#if NET7_0_OR_GREATER || CsWinRT_LANG_11_FEATURES
+            ReadOnlySpan<byte> functionName = "DllGetActivationFactory"u8;
+#else
+            string functionName = "DllGetActivationFactory";
+#endif
+            getActivationFactory = Platform.TryGetProcAddress(moduleHandle, functionName);
+            if (getActivationFactory == null)
+            {
+                module = null;
+                return false;
+            }
+
+            module = new DllModule(
+                fileName,
+                moduleHandle,
+                getActivationFactory);
+            return true;
+        }
+
+        private DllModule(string fileName, IntPtr moduleHandle, void* getActivationFactory)
+        {
+            _fileName = fileName;
+            _moduleHandle = moduleHandle;
+            _GetActivationFactory = (delegate* unmanaged[Stdcall]<IntPtr, IntPtr*, int>)getActivationFactory;
+
+            void* canUnloadNow = null;
+#if NET7_0_OR_GREATER || CsWinRT_LANG_11_FEATURES
+            ReadOnlySpan<byte> functionName = "DllCanUnloadNow"u8;
+#else
+            string functionName = "DllCanUnloadNow";
+#endif
+            canUnloadNow = Platform.TryGetProcAddress(_moduleHandle, functionName);
+
+            if (canUnloadNow != null)
+            {
+                _CanUnloadNow = (delegate* unmanaged[Stdcall]<int>)canUnloadNow;
+            }
+        }
+
+        public unsafe (ObjectReference<IActivationFactoryVftbl> obj, int hr) GetActivationFactory(string runtimeClassId)
+        {
+            IntPtr instancePtr = IntPtr.Zero;
+            try
+            {
+                MarshalString.Pinnable __runtimeClassId = new(runtimeClassId);
+                fixed (void* ___runtimeClassId = __runtimeClassId)
+                {
+                    int hr = _GetActivationFactory(MarshalString.GetAbi(ref __runtimeClassId), &instancePtr);
+                    if (hr == 0)
+                    {
+                        var objRef = ObjectReference<IActivationFactoryVftbl>.Attach(ref instancePtr);
+                        return (objRef, hr);
+                    }
+                    else
+                    {
+                        return (null, hr);
+                    }
+                }
+            }
+            finally
+            {
+                MarshalInspectable<object>.DisposeAbi(instancePtr);
+            }
+        }
+
+        ~DllModule()
+        {
+            System.Diagnostics.Debug.Assert(_CanUnloadNow == null || _CanUnloadNow() == 0); // S_OK
+            lock (_cache)
+            {
+                _cache.Remove(_fileName);

```

Comment:
How can this finalizer ever be triggered, if a loaded `DllModule` instance is stored in the `_cache` dictionary, which contains a strong reference to the `DllModule` itself? I don't see how any of those can ever be collected 🤔

Reply:
This was copied from the projection generated code without doing cleanup. It's possible it isn't needed.

Reply:
@manodasanW?

Reply:
I do believe this code path is unused on .NET 6+ as .NET doesn't run down all the finalizers when you exit an app.  But I have seen it get executed in our .NET Standard projection when running on .NET Framework which does.  The .NET GC can detect this circular reference and thereby I believe the finalizer would still run.  But we can probably run into unexpected results accessing the managed object in a finalizer as it probably has been collected.  But it is probably not finalized yet which is why we haven't heard folks running into issues from it.  For this change, I would leave this untouched and open a separate issue to track looking more into this separately as I do think there is some issues here from a perspective of .NET framework.

Reply:
> ".NET GC can detect this circular reference"

Wait, yes that is correct, but the `_cache` field is static here 😅

Reply:
Don't think this should be resolved, I'm still not sure this is correct. That field really is static.

Reply:
Let's do cleanup separate from this PR.